repl/src/main/scala/ammonite/repl/interp/Interpreter.scala

```diff
@@ -94,46 +103,52 @@ class Interpreter(prompt0: Ref[String],
     replApi
   )
   Timer("Interpreter init eval")
-  init()
-  Timer("Interpreter init init")
+
   val argString = replArgs.zipWithIndex.map{ case (b, idx) =>
     s"""
     val ${b.name} =
       ammonite.repl.frontend.ReplBridge.repl.replArgs($idx).value.asInstanceOf[${b.typeTag.tpe}]
     """
   }.mkString("\n")
 
-  val predefs = Seq(
-    (hardcodedPredef, Name("HardcodedPredef")),
-    (predef, Name("Predef")),
-    (storage.loadPredef, Name("LoadedPredef")),
-    (argString, Name("ArgsPredef"))
-  )
+  var predefImports = Imports(Nil)
 
-  // Use a var and a for-loop instead of a fold, because when running
-  // `processModule0` user code may end up calling `processModule` which depends
-  // on `predefImports`, and we should be able to provide the "current" imports
-  // to it even if it's half built
-  var predefImports = Imports()
-  for( (sourceCode, wrapperName) <- predefs) {
-    val pkgName = Seq(Name("ammonite"), Name("predef"))
-    processModule0(ImportHook.Source.File(wd/"<console>"), sourceCode, wrapperName, pkgName, predefImports) match{
-      case Res.Success(imports) =>
-        predefImports = predefImports ++ imports
-      case Res.Failure(ex, msg) =>
-        ex match{
-          case Some(e) => throw new RuntimeException("Error during Predef: " + msg, e)
-          case None => throw new RuntimeException("Error during Predef: " + msg)
-        }
+    val predefs = Seq(
+      (hardcodedPredef, "HardcodedPredef", "ammonite.predef"),
+      (predef, "Predef", "ammonite.predef"),
+      (storage.loadPredef, "LoadedPredef", "ammonite.predef"),
+      (argString, "ArgsPredef", "ammonite.predef")
+    )
 
-      case Res.Exception(ex, msg) =>
-        throw new RuntimeException("Error during Predef: " + msg, ex)
+    // Use a var and a for-loop instead of a fold, because when running
+    // `processModule0` user code may end up calling `processModule` which depends
+    // on `predefImports`, and we should be able to provide the "current" imports
+    // to it even if it's half built
+  for ((sourceCode, wrapperName, pkgName) <- predefs if !sourceCode.isEmpty) {
+
+    processModule(
+      ImportHook.Source.File(wd / "<console>"),
+      sourceCode,
+      Name(wrapperName),
+      pkgName.split('.').map(Name(_))) match {
+            case Res.Success(imports) =>
+              predefImports = predefImports ++ imports
+            case Res.Failure(ex, msg) =>
+              ex match {
+                case Some(e) => throw new RuntimeException("Error during Predef: " + msg, e)
+                case None => throw new RuntimeException("Error during Predef: " + msg)
+              }
+
+            case Res.Exception(ex, msg) =>
+              throw new RuntimeException("Error during Predef: " + msg, ex)
+          }
     }
-  }
+
+  updateClassPath
```

Comment:
Does this need to be above the `eval.sess.save()`, instead of below it like the original `init()` call is? If not, we should leave it in the same place